### DISCUSSION TITLE
Soroban implementation of OpenZeppelin/MerkleProof.sol

### DISCUSSION THREAD

**Discussion by palinko91** [stellar/sorobanathon#59] [General]
Hello

I really liked the projects being built here, and I got inspired to do something also!

Lately, I've been checking out Merkle trees and how they're implemented on Ethereum, so I thought why not bring them into Soroban?

It can be very handy to handle bigger whitelist spots because we don't need to write many addresses into the smart contract and check if the function caller is on that list. We simply have to store the root only, make the website UI send the proof data, and have the contract check whether the leaf is part of the tree or not. The leaf in this case is our wallet address account ID, so that we can make sure that if someone else sends the right proof, he can't get "true" because he's not whitelisted.

You can find the repo here: https://github.com/palinko91/soroban-merkleproof

Have fun!

**Comment by achallagundla** [comment]
Oooh, Thank you for your submission! I will get back to you shortly after review!

**Comment by achallagundla** [comment]
Heya @palinko91  ,
Congratulations! Your contribution to Sorobanathon: First Light has been accepted! 🎉
Your submission may qualify for a reward of 1000 XLM! One last step: to receive that reward, please fill out this [form](https://stellarform.typeform.com/Sorobanathon) completely and accurately.

Please let us know if you have any questions. (You can also message me on discord @ anuxhya)

All the best,
Anuhya
